src/service/chat.js

```diff
@@ -28,6 +28,47 @@ const createChannel = async (channelType, members, channelName) => {
   }
 };
 
+const followClient = async (members, data, text, textOwnMessage) => {
+  try {
+    const token = TokenStorage.get(ITokenEnum.token);
+    const id = await getUserId();
+    const user = {
+      id
+    };
+    const sort = [{last_message_at: -1}];
+    const filter = {type: 'messaging', members: {$eq: members}};
+    await chatClient.disconnectUser();
+    await chatClient.connectUser(user, token);
+    const channel = await chatClient.queryChannels(filter, sort, {
+      watch: true,
+      state: true
+    });
+    const message = {
+      user_id: data.user_id_follower,
+      text,
+      isSystem: true,
+      silent: true,
+      textOwnMessage,
+      userIdFollowed: data?.user_id_followed,
+      userIdFollower: data?.user_id_follower
+    };
+
+    const name = [data?.username_followed, data?.username_follower].join(',');
+
+    if (channel?.length <= 0) {
+      const newChannel = await createChannel('messaging', members, name);
+      return newChannel.sendMessage(message, {skip_push: true});
+    }
+    const messageClient = chatClient.channel('messaging', channel[0].id);
+    await messageClient.sendMessage(message, {skip_push: true});
+
+    return channel;
+  } catch (error) {
+    console.log('error follow client', error);
+    throw error;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followClient` function seems to be doing a lot of things. It's connecting the user, querying channels, sending messages, and creating channels if they don't exist. This is a violation of the Single Responsibility Principle (SRP). Consider breaking this function into smaller ones, each with its own responsibility. Also, it's not clear what happens when `TokenStorage.get(ITokenEnum.token)` or `getUserId()` fail. There should be error handling for these cases.

```diff
- const followClient = async (members, data, text, textOwnMessage) => {
+ const connectUser = async () => {
+   const token = TokenStorage.get(ITokenEnum.token);
+   const id = await getUserId();
+   const user = { id };
+   await chatClient.disconnectUser();
+   await chatClient.connectUser(user, token);
+ }
+
+ const queryChannels = async (members) => {
+   const sort = [{last_message_at: -1}];
+   const filter = {type: 'messaging', members: {$eq: members}};
+   return await chatClient.queryChannels(filter, sort, { watch: true, state: true });
+ }
+
+ const sendMessage = async (channel, data, text, textOwnMessage) => {
+   const message = {
+     user_id: data.user_id_follower,
+     text,
+     isSystem: true,
+     silent: true,
+     textOwnMessage,
+     userIdFollowed: data?.user_id_followed,
+     userIdFollower: data?.user_id_follower
+   };
+   const name = [data?.username_followed, data?.username_follower].join(',');
+   if (channel?.length <= 0) {
+     const newChannel = await createChannel('messaging', members, name);
+     return newChannel.sendMessage(message, {skip_push: true});
+   }
+   const messageClient = chatClient.channel('messaging', channel[0].id);
+   await messageClient.sendMessage(message, {skip_push: true});
+   return channel;
+ }
+
+ const followClient = async (members, data, text, textOwnMessage) => {
+   try {
+     await connectUser();
+     const channels = await queryChannels(members);
+     return await sendMessage(channels, data, text, textOwnMessage);
+   } catch (error) {
+     console.log('error follow client', error);
+     throw error;
+   }
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->